lib/src/iterable_matchers.dart

```diff
@@ -206,6 +215,26 @@ class _UnorderedMatches extends Matcher {
   Description describeMismatch(item, Description mismatchDescription,
           Map matchState, bool verbose) =>
       mismatchDescription.add(_test(item));
+
+  /// Returns [true] if the value at [valueIndex] can be paired with some
+  /// unmatched matcher.
+  ///
+  /// Recursively looks for new pairings whenever there is a conflict. [seen]
+  /// tracks the matchers that have already been consumed within this search.
+  bool _findPairing(List<List<bool>> adjacency, int valueIndex, Set<int> seen,
```

Comment:
nit: consider making `seen` optional and then doing `seen ??= new Set<int>();` so you don't have to pass in a new one when you call this method originally

Reply:
Done